### DISCUSSION TITLE
Queries aborted without a reason when using `client.query()`

### DISCUSSION THREAD

**Discussion by zorzysty** [urql-graphql/urql#3711] [Q&A] [answered]
When using `client.query`, running two requests with the same query, but different variables sometimes leads to the second request being cancelled.

```ts
  const response = await client
    .query(query, variables, {
      requestPolicy: "network-only",
      fetchOptions: { credentials: 'include' },
    })
    .toPromise()
```

It works fine most of the time, but every so often it happens that the second request using the same query with different variables that's called immediately after the first one, gets cancelled, and we're getting this logged out:

![image](https://github.com/user-attachments/assets/b5603b56-5593-4775-b9cc-5b14324fd73f)

We're using the following exchanges:
```ts
devtoolsExchange,
requestPolicyExchange({ ttl: Infinity }),
cacheExchange(...),
authExchange(...),
fetchExchange
```

What could be the reason for this cancellation? How can we avoid that?

**Comment by kitten** [comment, marked as answer]
This doesn't really seem to entirely be an issue in the core library? I can spot a reference to `fetchWithAbort` in the logs that seems to not handle the abort error correctly. The question then obviously also becomes why the request is being aborted, but that wrapper seems to be a bit suspicious potentially?
